tests/databases/test_postgres.py

```diff
@@ -320,3 +320,137 @@ def test_create_table_from_select_statement(database_table_fixture):
     expected = pd.DataFrame([{"id": 1, "name": "First"}])
     test_utils.assert_dataframes_are_equal(df, expected)
     database.drop_table(target_table)
+
+
+@pytest.mark.parametrize(
+    "parameters",
+    [
+        {
+            "merge_cols": ["sell"],
+            "source_tables_cols": ["sell", "list"],
+            "target_tables_cols": ["sell", "list"],
+            "conflict_strategy": "update",
+            "expected_df": pd.DataFrame(
+                [
+                    {"sell": 142, "list": 160},
+                    {"sell": 162, "list": 330},
+                    {"sell": 175, "list": 600},
+                ]
+            ),
+        },
+        {
+            "merge_cols": ["sell"],
+            "source_tables_cols": ["sell", "list"],
+            "target_tables_cols": ["sell", "list"],
+            "conflict_strategy": "ignore",
+            "expected_df": pd.DataFrame(
+                [
+                    {"sell": 142, "list": 160},
+                    {"sell": 162, "list": 330},
+                    {"sell": 175, "list": 540},
+                ]
+            ),
+        },
+        # {
+        #     'merge_cols': ['sell'],
+        #     'source_tables_cols': ['sell'],
+        #     'target_tables_cols': ['sell'],
+        #     'conflict_strategy': 'ignore',
+        #     'expected_df': pd.DataFrame([
+        #                         {'sell': 142, 'list': 160},
+        #                         {'sell': 162, 'list': None},
+        #                         {'sell': 175, 'list': 540}
+        #                     ])
+        # }
+    ],
+)
+def test_merge_table(parameters):
+    target_df = pd.DataFrame([{"sell": 142, "list": 160}, {"sell": 175, "list": 540}])
+    source_df = pd.DataFrame([{"sell": 162, "list": 330}, {"sell": 175, "list": 600}])
+
+    target_table = Table(conn_id="postgres_conn")
+    source_table = Table(conn_id="postgres_conn")
+
+    merge_cols = parameters["merge_cols"]
+
+    db = create_database(conn_id=target_table.conn_id)
+    db.load_pandas_dataframe_to_table(
+        source_dataframe=target_df, target_table=target_table
+    )
+    db.load_pandas_dataframe_to_table(
+        source_dataframe=source_df, target_table=source_table
+    )
+
+    db.run_sql(
+        sql_statement=f"ALTER TABLE {db.get_table_qualified_name(target_table)} "
+        f"ADD CONSTRAINT con_{target_table.name} UNIQUE ({','.join(merge_cols)})"
+    )
```

Comment:
How are we tearing down this constraint?

Reply:
Used fixture to teardown the table created.